vendor/calcite-1_22_0/build.gradle

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins { id 'org.apache.beam.vendor-java' }
+
+description = "Apache Beam :: Vendored Dependencies :: Calcite 1.22.0"
+
+group = "org.apache.beam"
+version = "0.1"
+
+def calcite_version = "1.22.0"
+def avatica_version = "1.16.0"
+def prefix = "org.apache.beam.vendor.calcite.v1_22_0"
+
+List<String> packagesToRelocate = [
+        "antlr",
+        "com.beust",
+        "com.esri",
+        "com.fasterxml",
+        "com.google.common",
+        "com.google.gson",
+        "com.google.inject",
+        "com.google.protobuf",
+        "com.google.thirdparty",
+        "com.jayway.jsonpath",
+        "com.yahoo",
+        "javax",
+        "net.minidev",
+        "net.sf",
+        "org.antlr",
+        "org.aopalliance",
+        "org.apache.calcite",
+        "org.apache.commons",
+        "org.apache.http",
+        "org.apache.tool",
+        "org.apiguardian.api",
+        "org.codehaus",
+        "org.easymock",
+        "org.eclipse.jetty",
+        "org.json",
+        "org.objectweb",
+        "org.objenesis",
+        "org.pentaho",
+        "org.slf4j",
+        "org.testng",
+        "org.yaml",
+]
+
+vendorJava(
+        dependencies: [
+                "org.apache.calcite:calcite-core:$calcite_version",
+                "org.apache.calcite:calcite-linq4j:$calcite_version",
+                "org.apache.calcite.avatica:avatica-core:$avatica_version",
+                library.java.protobuf_java,
+                library.java.slf4j_api,
+
+                "cglib:cglib:3.3.0",
+                "com.google.code.gson:gson:2.8.6",
+                "javax.servlet:javax.servlet-api:3.0.1",
+                "javax.transaction:javax.transaction-api:1.2",
+                "org.antlr:stringtemplate:3.2.1",
+                "org.apache.ant:ant:1.9.2",
+                "org.codehaus.jettison:jettison:1.4.0",
+                "org.easymock:easymock:4.1",
+                "org.eclipse.jetty:jetty-jmx:9.4.27.v20200227",
+                "org.json:json:20190722",
+                "org.testng:testng:7.1.0",
+        ],
+        runtimeDependencies: [
+                "com.jayway.jsonpath:json-path:2.4.0",
```

Comment:
Why are you relocating com.jayway and marking json-path as a runtime dependency?

Ditto for other libraries below such as commons-logging, antlr, slf4j

Reply:
move antlr to dependencies and relocate it and not relocate commons-logging and slf4j